### DISCUSSION TITLE
React Instantsearch SSR on Cloudflare Workers

### DISCUSSION THREAD

**Discussion by matheusgrieger** [algolia/instantsearch#5325] [Q&A]
Hello!

We are trying to migrate our current PHP website to React using Shopify Hydrogen, and we're hosting it on Cloudflare Workers. While it was possible to get SSR working with `react-instantsearch-hooks-server` following the guide on the website when running the development server (Node.js), it is crashing on Cloudflare Workers and Miniflare used for testing.

I know the current implementation uses Node HTTP to get the initial results, but is it at all possible to switch to using fetch, which is available on edge like Workers?

**Comment by francoischalifour** [comment]
It should work with `algoliasearch` [v4.14.0](https://github.com/algolia/algoliasearch-client-javascript/releases/tag/4.14.0) which introduced support for Cloudflare Workers and Edge Functions. Could you upgrade this dependency and let us know?

**Reply by matheusgrieger** [reply]
It still doesn't work. I've tried upgrading `algoliasearch` to v4.14.1, and adding `@algolia/request-fetcher`'s `createFetchRequester` to the search client `requester` option, but none of that did the trick. Am I missing something?

**Reply by Haroenv** [reply]
I don't have a cloudflare worker setup, but did you do this?

```js
import algoliasearch from 'algoliasearch/lite';
import { createFetchRequester } from '@algolia/requester-fetch';

const searchClient = algoliasearch('...', '...', { requester: createFetchRequester() })

// ...
<InstantSearch searchClient={searchClient}>
```

**Reply by matheusgrieger** [reply]
Yes, that's what I did. Wrapped the `InstantSearch` component with `InstantSearchSSRProvider`, too.
